### DISCUSSION TITLE
best install method? (.deb/source)

### DISCUSSION THREAD

**Discussion by calebd-anderson** [inspircd/inspircd#1862] [Support]
There seems to be two install methods. **1)** from source `make install` **2)** `apt-get install` the latest pre-built .deb file.
I'm on Ubuntu 20.04 LTS and I normally install packages from the default repos. The version in the default repo for Ubuntu is obsolete and it didn't load correctly after reboot.
I did get a connection working to my server after installing from the pre-built package install method here https://docs.inspircd.org/3/installation/ubuntu/
The question is I get an "ERROR: Cannot open config file: /etc/inspircd/inspircd.conf" when I use the bin command `inspircd`.
It looks like most tutorials install from source or maybe create a dedicated unprivileged user. Are the source files then supposed to remain in the home directory after make install? _I got a little lost_ there after installing from the pre-built package (.deb file) in the home directory. I also got this error when installing from the pre-built pacakge.
"N: Download is performed unsandboxed as root as file '/home/*removed*/inspircd_3.9.0.ubuntu20.04.2_amd64.deb' couldn't be accessed by user '_apt'. - pkgAcquire::Run (13: Permission denied)"
What is the correct way to install inspircd with correct permissions on everything?
Thanks

**Comment by sadiepowell** [comment]
>I got a connection to my server working with the pre-built package install method here https://docs.inspircd.org/3/installation/ubuntu/
The question is I get an "ERROR: Cannot open config file: /etc/inspircd/inspircd.conf" when I use the bin command inspircd.

You actually need to create a config file for it. If you read /etc/inspircd/help.txt it should tell you where you can find the example configs that can be used as a basis for your config file.

>It looks like most tutorials install from source or maybe create a dedicated unprivileged user. Are the source files then supposed to remain in the home directory after make install? I got a little lost there after installing from the pre-built package in the home directory.

If you're building from source you can delete the source files once built but it doesn't really matter either way.

>I also got this error when installing from the pre-built pacakge.

>"N: Download is performed unsandboxed as root as file '/home/removed/inspircd_3.9.0.ubuntu20.04.2_amd64.deb' couldn't be accessed by user '_apt'. - pkgAcquire::Run (13: Permission denied)"

I've never seen this before. How did you download and install it?

**Comment by calebd-anderson** [comment]
I downloaded with `wget https://github.com/inspircd/inspircd/releases/download/v3.9.0/inspircd_3.9.0.ubuntu20.04.2_amd64.deb`
to my sudo user home directory then
`sudo apt install ./inspircd_3.9.0.ubuntu20.04.2_amd64.deb`

Thanks @SadieCat, that's allot more clear now. Okay, I see I had removed _other read and execute_ from my home directory so I `chmod o+rx <home_dir>` and that seems to have stopped the "N: Download ... unsandboxed" error thing. The N: prefix must mean something; I've seen it other places with W: and E: but I'm not sure what it indicates. Just to be clear that was an error that appeared in the last record after running `sudo apt install ./inspircd_3.9.0.ubuntu20.04.2_amd64.deb` not from initially downloading the package.

Now when I install - the help file isn't going to /etc/inspircd/ so the directory is made but it's _empty_. I'm not sure what permission thing is going wrong here. I'm comparing it to a VM and it looks correct.
Last thing, in the VM now I get "Failed to write PID-file '/var/run/inspircd.pid', exiting." instead of "can't open config file".

**Reply by calebd-anderson** [reply]
> Now when I install; the help file isn't going to /etc/inspircd/ so the directory is made but it's empty. I'm not sure what permission thing is going wrong here. I'm comparing it to a VM and it looks correct.

Okay, so what I was doing was I manually removed `sudo rm -r /etc/inspircd/` and what I had to do was `sudo apt purge inspircd` then `sudo apt install ./inspircd_3.9.0.ubuntu20.04.2_amd64.deb` and the help file came back.

**Comment by calebd-anderson** [comment]
It must have been been the permissions on "/etc/inspircd/" why `inspircd` bin command was saying it couldn't open the config file. That seems to be fixed now.

I still get "Failed to write PID-file '/var/run/inspircd.pid', exiting." however.

Thanks

**Reply by calebd-anderson** [reply]
> I still get "Failed to write PID-file '/var/run/inspircd.pid', exiting." however.

I see, I can pass the option `--nopid`. I don't necessarily understand the implications of that though.

**Reply by sadiepowell** [reply]
You should probably start InspIRCd with systemd (`systemctl start inspircd.service`) rather than running the executable manually as its preconfigured to work with the system-wide install.

**Comment by calebd-anderson** [comment]
If I've installed from the .deb file how do I:
- get "m_ssl_openssl.so" to load
- Apparently there is no `.configure` available if installed from the .deb file

**Comment by sadiepowell** [comment]
The ssl_openssl module can not legally be shipped in binary form due to license conflicts. You should use the ssl_gnutls module instead.
